packages/tiled/lib/src/objects/tiled_object.dart

```diff
@@ -59,7 +59,7 @@ class TiledObject {
   bool point;
   bool rectangle;
 
-  Template? template;
+  Future<Template?>? template;
```

Comment:
Why does this need to be a future?

Reply:
The template object must wait for the template file to be loaded.
We also can’t load it beforehand because we can’t know which file to load until reaching a template property.

Reply:
But it will be loaded while parsing right? Since this is mutable for some reason it could just be set once it is finished?

Reply:
Yes it would be the easier way. However, all the call stack down to TiledObject.parse is synchronous, if we make it async, we must change the upstream too. I want to avoid a breaking change.

Reply:
But since it is mutable you don't have to make it async, just set it when it is done?

Reply:
Hmm, this new TiledObject instance is added to the final parsed object after `load()` function is completed. If the `template` is mutated after that, users can't know when it happens to get the latest value. If using Future, at least they know explicitly that they need to `await` it to get the value. That's my intention. But I have to admit that it's not the best solution.
I've just thought about another solution. If `load()` function somehow can await for all pending Futures complete, it's the best.

Reply:
> I've just thought about another solution. If `load()` function somehow can await for all pending Futures complete, it's the best.

Yeah, that sounds much better, that shouldn't be a problem to do.